src/requests.js

```diff
@@ -0,0 +1,342 @@
+    for (let i = 0; i < headers.length; i++) {
+        const element = headers[i];
+        if (skipParams.includes(element.name.toLowerCase())) continue;
+        var param = {
+            'name': element.name,
+            'in': 'header',
+            'description': `e.g. ${element.value}`,
+            'required': false
+        };
+        if (context.options.openApi3) {
+            param.schema = { type: 'string' }
+            param.example = element.value
+        } else {
+            param['x-example'] = element.value
+            param.type = 'string' // TODO: string, number, boolean, integer, array
+        }
+        if (!params.find(p => isEqual(p, param)))
+        params.push(param);
+    }
+    return params;
+};
+
+// generate security and securityDefinitions from authorization headers
+function swaggerSecurity(context, headers) {
+    var security = null;
+    headers.filter(function (header) {
+        return header.name.toLowerCase() === 'authorization';
+    }).forEach(function (header) {
+        if (header.value.match(/^Basic /)) {
+            if (!security) security = {};
+            security['basic'] = [];
+            context.swagger.securityDefinitions['basic'] = { type: 'basic' };
+        } else if (header.value.match(/^Bearer /)) {
+            if (!security) security = {};
+            if (context.options.bearerAsApikey) {
+                security['bearer'] = [];
+                context.swagger.securityDefinitions['bearer'] = {
+                    type: 'apiKey', in: 'header', name: 'Authorization'
+                };
+            } else {
+                security['oauth2'] = [];
+                context.swagger.securityDefinitions['oauth2'] = {
+                    type: 'oauth2', flow: 'accessCode',
+                    authorizationUrl: '', tokenUrl: '', scopes: {}
+                };
+            }
+        }
+    });
+    return security;
+}
+
+const setSecurity = (context, request, operation) => {
+    if (context.options.openApi3){
+        return operation
+    }
+    var security = swaggerSecurity(context, request.headers);
+    if (security) {
+        if (!operation.security) {
+            operation.security = [security];
+        } else {
+            if (!operation.security.find(s => isEqual(s, security))){
+                operation.security.push(security);
+            }
+        }
+    }
+    return operation
+}
+
+const setSwaggerRequestSchema = (operation, schema) => {
+    if (!schema.length) {
+        return operation
+    }
+    if (schema.length === 1) {
+        operation.parameters.push({ name: 'body', in: 'body', schema: schema[0].scheme })
+        return operation
+    } 
+    operation.parameters.push({ 
+        name: 'body', 
+        in: 'body', 
+        schema: { 
+            anyOf: schema.map((s) => s.scheme)
+        } 
+    });
+    return operation
+}
+
+const mergeHeaders = (headers, parameters, openApi3) => {
```

Comment:
Instead of explicitly passing in `openapi3`, let's use the `options` variable or `context` to derive the state